apiserver/common/charmhub.go

```diff
@@ -0,0 +1,34 @@
+// Copyright 2020 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package common
+
+import (
+	"github.com/juju/errors"
+
+	"github.com/juju/juju/charmhub"
+	"github.com/juju/juju/state"
+)
+
+// createCharmhubClient creates a new charmhub Client based on this model's
+// config.
+func CharmhubClient(st *state.State) (*charmhub.Client, error) {
```

Comment:
Thanks. I intend to do this, however per discussion at our standup last week, this will be a fair bit of refactoring, so we want to get this PR over the goal line and refactor in a subsequent PR.